addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/ZoneControlXML.java

```diff
@@ -0,0 +1,205 @@
+     */
+    @Override
+    public void setVolumeRelative(ZoneControlState state, float percent)
+            throws IOException, ReceivedMessageParseException {
+        setVolume(state.volume + percent);
+    }
+
+    @Override
+    public void setMute(boolean mute) throws IOException, ReceivedMessageParseException {
+        if (mute) {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Volume><Mute>On</Mute></Volume>"));
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Volume><Mute>Off</Mute></Volume>"));
+        }
+        update();
+    }
+
+    @Override
+    public void setInput(String name) throws IOException, ReceivedMessageParseException {
+        comReference.get().send(XMLUtils.wrZone(zone, "<Input><Input_Sel>" + name + "</Input_Sel></Input>"));
+        update();
+    }
+
+    @Override
+    public void setSurroundProgram(String name) throws IOException, ReceivedMessageParseException {
+        if (name.toLowerCase().equals("straight")) {
+            comReference.get().send(XMLUtils.wrZone(zone,
+                    "<Surround><Program_Sel><Current><Straight>On</Straight></Current></Program_Sel></Surround>"));
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Surround><Program_Sel><Current><Sound_Program>" + name
+                    + "</Sound_Program></Current></Program_Sel></Surround>"));
+        }
+
+        update();
+    }
+
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        if (observer == null) {
+            return;
+        }
+
+        logger.trace("Updating status");
+
+        AbstractConnection com = comReference.get();
+        String response = com.sendReceive(XMLUtils.wrZone(zone, "<Basic_Status>GetParam</Basic_Status>"));
+        Document doc = XMLUtils.xml(response);
+        if (doc.getFirstChild() == null) {
+            throw new ReceivedMessageParseException("<Basic_Status>GetParam failed: " + response);
+        }
+        Node basicStatus = XMLUtils.getNode(doc.getFirstChild(), zone + "/Basic_Status");
+
+        Node node;
+        String value;
+
+        ZoneControlState state = new ZoneControlState();
+
+        node = XMLUtils.getNode(basicStatus, "Power_Control/Power");
+        value = node != null ? node.getTextContent() : "";
+        state.power = "On".equalsIgnoreCase(value);
+
+        node = XMLUtils.getNode(basicStatus, "Input/Input_Sel");
+        value = node != null ? node.getTextContent() : "";
+        state.inputID = XMLUtils.convertNameToID(value);
+
+        if (StringUtils.isBlank(state.inputID)) {
+            throw new ReceivedMessageParseException(
+                    "Expected inputID. Failed to read Input/Input_Sel_Item_Info/Src_Name");
+        }
+
+        // Some receivers may use Src_Name instead?
+        node = XMLUtils.getNode(basicStatus, "Input/Input_Sel_Item_Info/Title");
+        value = node != null ? node.getTextContent() : "";
```

Comment:
@martinvw  Yes, I actually refactored some of this in my other branch ([2574_yamaha_spotify](https://github.com/zarusz/openhab2-addons/tree/2574_yamaha_spotify)) that will add Spotify feature #2574. I intended to merge it to this PR. Do you suggest I merge to this PR or you could wait with this until another PR ?